client/setec/store.go

```diff
@@ -227,10 +253,12 @@ func (s *Store) secretLocked(name string) Secret {
 	f, ok := s.active.f[name]
 	if !ok {
 		f = func() []byte {
-			s.active.RLock()
-			defer s.active.RUnlock()
+			s.active.Lock()
```

Comment:
If we now have to Lock on every access, does the RWLock still have any value? Should we use a regular mutex?

Reply:
It's probably not strictly necessary anymore. We still use it to grab a snapshot during the poll, and when flushing the cache (if defined). But the long-running case during the poll was the main reason for it, and that's now split up.

Reply:
(removed)